src/Application/Providers/AlarmsConditions/RespondToAlarmEventsProvider.cs

```diff
@@ -0,0 +1,89 @@
+﻿// SPDX-License-Identifier: MIT. 
+// Copyright Contributors to the Open Manufacturing Platform.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OMP.Connector.Application.Providers.AlarmSubscription.Base;
+using OMP.Connector.Domain.Configuration;
+using OMP.Connector.Domain.Extensions;
+using OMP.Connector.Domain.Schema.Enums;
+using OMP.Connector.Domain.Schema.Request.AlarmSubscription;
+using OMP.Connector.Domain.Schema.Responses.AlarmSubscription;
+using Opc.Ua;
+using Opc.Ua.Client;
+
+namespace OMP.Connector.Application.Providers.AlarmSubscription
+{
+    public class RespondToAlarmEventsProvider : AlarmSubscriptionProvider<RespondToAlarmEventsRequest, RespondToAlarmEventsResponse>
+    {
+        public RespondToAlarmEventsProvider(
+            ILogger<RespondToAlarmEventsProvider> logger,
+            IOptions<ConnectorConfiguration> connectorConfiguration,
+            RespondToAlarmEventsRequest command) : base(command, connectorConfiguration, logger){}
+
+        protected override async Task<string> ExecuteCommandAsync()
+        {
+            var results = new List<string>();
+            foreach(var eventAction in this.Command.AlarmEventActions)
+            {
+                var methodNodeId = eventAction.Action switch
+                {
+                    "Acknowledge" => MethodIds.AcknowledgeableConditionType_Acknowledge,
+                    "Confirm" => MethodIds.AcknowledgeableConditionType_Confirm,
+                    _ => null
+                };
+                var statusCode = CallMethod(eventAction.SourceNodeId, methodNodeId, eventAction.Comment, eventAction.EventId);
+                if(StatusCode.IsBad(statusCode))
+                    results.Add(GetBadCallResult(statusCode, eventAction.Action, eventAction.Comment));
+            }
+
+            this.Logger.Debug($"Created/Updated alarm subscriptions on Endpoint: [{this.EndpointUrl}]");
+
+            return this.GetStatusMessage(results);
```

Comment:
Can we rather return Task.FromResult(... so we don net get this warning:
This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking 
Or is there a reason why we opted fir this return?

Reply:
I removed the async modifier and used the FromResult as suggested.